main.lua

```diff
@@ -249,31 +279,49 @@ if REPENTANCE then
 			end
 		end
 		
-		-- save last real collectible. for Flip Item handling
-		if decrease then
-			lastGetItemResult = {selectedCollectible, curFrame, nil}
-		end
+		initialItemNext = false
+		flipItemNext = false
 	end
 	EID:AddCallback(ModCallbacks.MC_POST_GET_COLLECTIBLE, EID.postGetCollectible)
 
 	-- Handle Flip Item spawn
+	function EID:preRoomEntitySpawn(entityType, variant, subtype, gridIndex, seed)
+		flipItemNext = false
+		if entityType == 5 and (variant == 100 or variant == 150) then
+			lastGetItemResult = {nil, Isaac.GetFrameCount(), gridIndex}
+			initialItemNext = true
+		end
+	end
+	EID:AddCallback(ModCallbacks.MC_PRE_ROOM_ENTITY_SPAWN, EID.preRoomEntitySpawn)
+	
 	function EID:postPickupInit(entity)
-		local curFrame = Isaac.GetFrameCount()
-		if entity.SubType == lastGetItemResult[1] and curFrame == lastGetItemResult[2] then
-			local curRoomIndex = game:GetLevel():GetCurrentRoomIndex()
-			if EID.flipItemPositions[curRoomIndex] == nil then
-				EID.flipItemPositions[curRoomIndex] = {}
-			end
-			local gridPos = game:GetRoom():GetGridIndex(entity.Position)
-			local flipEntry = EID.flipItemPositions[curRoomIndex][gridPos]
-			if not flipEntry then
-				lastGetItemResult[3] = gridPos
-			else
-				entity:GetData()["EID_FlipItemID"] = flipEntry
+		flipItemNext = true
+		
+		-- Keep the Flip item the same for pedestal rerolls
+		local curRoomIndex = game:GetLevel():GetCurrentRoomIndex()
+		local gridPos = game:GetRoom():GetGridIndex(entity.Position)
+		local flipEntry = EID.flipItemPositions[curRoomIndex] and EID.flipItemPositions[curRoomIndex][gridPos]
+		if flipEntry then entity:GetData()["EID_FlipItemID"] = flipEntry end
+	end
+	EID:AddCallback(ModCallbacks.MC_POST_PICKUP_INIT, EID.postPickupInit)
```

Comment:
Add the optional attribute `PickupVariant.PICKUP_COLLECTIBLE` to the AddCallback line, in order for this function to be called less often.